src/Usos/README.md

```diff
@@ -0,0 +1,65 @@
+# Usos
+
+```bash
+composer require socialiteproviders/usos
+```
+
+## Installation & Basic Usage
+
+Please see the [Base Installation Guide](https://socialiteproviders.com/usage/), then follow the provider specific instructions below.
+
+### Add configuration to `config/services.php`
+
+```php
+'usos' => [
+  'domain' => env('USOS_DOMAIN_URL'), // Because of every instance of USOS is self-hosted
+  'client_id' => env('USOS_CLIENT_ID'),
+  'client_secret' => env('USOS_CLIENT_SECRET'),
+  'redirect' => env('USOS_REDIRECT_URI')
+],
+```
+
+As a default, provider loads only necessary for Socialite data. If you need to get more data during authentication process, such as student status or phone number, you may add ```profile_fields_selector``` parameter:
+
+```php
+'usos' => [
+  // ...
+  'profile_fields_selector' => env('USOS_PROFILE_FIELDS_SELECTOR'),
```

Comment:
Don't see where this is used?

Reply:
According to the [USOS API documentation](https://apps.usos.edu.pl/developers/api/services/users/#user), only ID, first name and last name will be returned, if 'fields' selector wasn't provided.

According to the [Laravel Socialite Docs](https://laravel.com/docs/9.x/socialite#retrieving-user-details) Socialite driver must provide this set of data: id, name, email, and avatar. Email address and avatar (user's photo) won't be returned by API endpoint by default, so request must contain 'fields' parameter to get all necessary data (minimum: id|first_name|last_name|email|photo_urls). 

Because of the large list of possible fields, which could be returned, I've decided to allow application to re-define the list of extra parameters, which application need to get. For example: authentication only could be possible for students. In this situation, application can ask about a student status of user (student_status field from selector).

UPD: Sorry, I forgot to include this change to Server.php. I'll add it in the next commit

Reply:
Haha yea I didn't see it used in the code so was confused. Thanks!

Reply:
I've fixed it in the next commit. It was automatically attached to this PR. If you let me, I'll mark this as a `resolved`